tests/personal_data/portfolios/types/test_future_portfolio.py

```diff
@@ -14,150 +14,690 @@
+        contract_type=enums.FutureContractType.INVERSE_PERPETUAL,
+        current_leverage=DEFAULT_FUTURE_SYMBOL_LEVERAGE)
+    trader_inst.exchange_manager.exchange.set_pair_future_contract(DEFAULT_FUTURE_SYMBOL, symbol_contract)
 
     # Test buy order
-    market_buy = BuyMarketOrder(trader)
-    market_buy.update(order_type=TraderOrderType.SELL_MARKET,
-                      symbol="BTC/USDT",
+    market_buy = BuyMarketOrder(trader_inst)
+    market_buy.update(order_type=enums.TraderOrderType.BUY_MARKET,
+                      symbol=DEFAULT_FUTURE_SYMBOL,
                       current_price=decimal.Decimal(str(1000)),
-                      quantity=decimal.Decimal(str(10)),
+                      quantity=decimal.Decimal(str(1000)),  # real quantity = 1000 / 1000 = 1 BTC at 1000$
                       price=decimal.Decimal(str(1000)))
 
     # test buy order creation
     portfolio_manager.portfolio.update_portfolio_available(market_buy, True)
-    assert portfolio_manager.portfolio.get_currency_portfolio("BTC", PORTFOLIO_AVAILABLE) == decimal.Decimal(str(10))
-    assert portfolio_manager.portfolio.get_currency_portfolio("USDT", PORTFOLIO_AVAILABLE) == decimal.Decimal(str(995))
-    assert portfolio_manager.portfolio.get_currency_portfolio("BTC", PORTFOLIO_TOTAL) == decimal.Decimal(str(10))
-    assert portfolio_manager.portfolio.get_currency_portfolio("USDT", PORTFOLIO_TOTAL) == decimal.Decimal(str(1000))
+    assert portfolio_manager.portfolio.get_currency_portfolio("BTC").available == decimal.Decimal(str(9))
+    assert portfolio_manager.portfolio.get_currency_portfolio("BTC").total == decimal.Decimal(str(10))
 
+    # Test buy order
+    market_buy = BuyMarketOrder(trader_inst)
+    market_buy.update(order_type=enums.TraderOrderType.BUY_MARKET,
+                      symbol=DEFAULT_FUTURE_SYMBOL,
+                      current_price=decimal.Decimal(str(74)),
+                      quantity=decimal.Decimal(str(74 / 3)),  # real quantity = 0.3 BTC at 74$
+                      price=decimal.Decimal(str(74)))
 
-@pytest.mark.parametrize("backtesting_exchange_manager", [(None, DEFAULT_EXCHANGE_NAME, False, False, True)],
-                         indirect=["backtesting_exchange_manager"])
-async def test_update_portfolio_available_from_order(backtesting_trader):
-    config, exchange_manager, trader = backtesting_trader
-    portfolio_manager = exchange_manager.exchange_personal_data.portfolio_manager
-    await init_symbol_contract(exchange_manager.exchange, leverage=decimal.Decimal(10))
+    # test buy order creation
+    portfolio_manager.portfolio.update_portfolio_available(market_buy, True)
+    assert portfolio_manager.portfolio.get_currency_portfolio("BTC").total == decimal.Decimal(10)
+    # 9 - (74 / 3) / 74 = 8.666666666666666648648648649
+    assert portfolio_manager.portfolio.get_currency_portfolio("BTC").available == decimal.Decimal(
+        '8.666666666666666648648648649')
```

Comment:
this number won't make issues on different os architectures ?

Reply:
I don't know, you think it can be an issue even when using `decimal.Decimal` ?

Reply:
Right, as long as it is initialized as a str, there should be no problem

Reply:
Right I see, I'll fix it